tests/unit/util/BlockingCacheTests.cpp

```diff
@@ -0,0 +1,254 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2025, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include "util/AsioContextTestFixture.hpp"
+#include "util/Assert.hpp"
+#include "util/BlockingCache.hpp"
+#include "util/NameGenerator.hpp"
+
+#include <boost/asio/error.hpp>
+#include <boost/asio/post.hpp>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <memory>
+
+using testing::MockFunction;
+using testing::Return;
+using testing::StrictMock;
+
+#include <boost/asio/spawn.hpp>
+#include <boost/asio/steady_timer.hpp>
+
+#include <chrono>
+#include <expected>
+#include <string>
+
+struct BlockingCacheTest : SyncAsioContextTest {
+    using ErrorType = std::string;
+    using ValueType = int;
+    using Cache = util::BlockingCache<ValueType, ErrorType>;
+    using MockUpdater = StrictMock<MockFunction<std::expected<ValueType, ErrorType>(boost::asio::yield_context)>>;
+    using MockVerifier = StrictMock<MockFunction<bool(ValueType const&)>>;
+
+    std::unique_ptr<Cache> cache = std::make_unique<Cache>();
```

Comment:
I thought we decided we want members to be protected (and have an underscore at the end)?

Reply:
I thought we didn't come to a strict rule about it and both variants are acceptable. But we could choose a style.

Reply:
Prefer consistency if possible but overall you are not wrong 👍